src/librustc/traits/error_reporting.rs

```diff
@@ -794,6 +796,19 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
                             // If it has a custom `#[rustc_on_unimplemented]` note, let's display it
                             err.note(s.as_str());
                         }
+                        if let Some(ref s) = enclosing_scope {
+                            let enclosing_scope_span = tcx.def_span(
```

Comment:
I wonder if we should just highlight the function name (see below) or return type instead of the whole body - often these spans can be really long and I tend to find them hard to read. Thoughts, @rust-lang/wg-diagnostics?

https://github.com/rust-lang/rust/blob/ab2155750123d2b6858e521fcad8a6097d0777e4/src/libsyntax_pos/source_map.rs#L886-L893

Reply:
I normally agree, particularly because some IDEs will highlight the entire thing making the error borderline useless, but in cases similar to this, where knowing where the block _ends_ might be useful, I prefer this CLI output. We can of course merge as is and open a follow up ticket to explore the options around this.